src/main/java/org/wso2/carbon/inbound/salesforce/poll/SalesforceStreamData.java

```diff
@@ -254,36 +246,57 @@ private void loadMandatoryParameters(Properties properties) {
      * @param properties The Optional parameters of Salesforce.
      */
     private void loadOptionalParameters(Properties properties) {
-        if (log.isDebugEnabled()) {
-            log.debug("Starting to load the salesforce credentials");
+        if (LOG.isDebugEnabled()) {
+            LOG.debug("Starting to load the salesforce credentials");
         }
         if (properties.getProperty(SalesforceConstant.CONNECTION_TIMEOUT) == null) {
             connectionTimeout = SalesforceConstant.CONNECTION_TIMEOUT_DEFAULT;
         } else {
             try {
                 connectionTimeout = Integer.parseInt(properties.getProperty(SalesforceConstant.CONNECTION_TIMEOUT));
             } catch (NumberFormatException e) {
-                log.error("The Value should be in Number", e);
+                LOG.error("The Value should be in Number", e);
             }
         }
-
-        if (properties.getProperty(SalesforceConstant.READ_TIMEOUT) == null) {
-            readTimeout = SalesforceConstant.READ_TIMEOUT_DEFAULT;
+        SalesforceDataHolderObject.setConnectionTimeout(connectionTimeout);
+        if (properties.getProperty(SalesforceConstant.WAIT_TIME) == null) {
+            waitTime = SalesforceConstant.WAIT_TIME_DEFAULT;
         } else {
             try {
-                readTimeout = Integer.parseInt(properties.getProperty(SalesforceConstant.READ_TIMEOUT));
+                waitTime = Integer.parseInt(properties.getProperty(SalesforceConstant.WAIT_TIME));
             } catch (NumberFormatException e) {
-                log.error("The Value should be in Number", e);
+                LOG.error("The Value should be in Number", e);
             }
         }
-
-        if (properties.getProperty(SalesforceConstant.WAIT_TIME) == null) {
-            connectionTimeout = SalesforceConstant.WAIT_TIME_DEFAULT;
+        SalesforceDataHolderObject.setWaitTime(waitTime);
+        //replay enable
+        if (Boolean.parseBoolean(properties.getProperty(SalesforceConstant.REPLAY_FROM))) {
+
+            if (properties.getProperty(SalesforceConstant.REPLAY_FROM_ID_Stored_File_Path) != null && !StringUtils.
+                    isEmpty(properties.getProperty(SalesforceConstant.REPLAY_FROM_ID_Stored_File_Path))) {
+                try {
+                    //read id from file
+                    replayFromOption = readFromGivenFile(
```

Comment:
if number format exception happens here, does the replayFromOption value is set to a default value?